### PR TITLE
cluster_resolver: wait for child policy configuration update inline

### LINKED ISSUE #7210: clusterimpl: update picker synchronously upon receipt of configuration update
https://github.com/grpc/grpc-go/issues/5469 recommends an audit of existing LB policies to ensure that they update their pickers synchronously upon receipt of a configuration update.

`xds_cluster_impl_experimental` does not update its picker synchronously.

- Configuration updates are handled here: https://github.com/grpc/grpc-go/blob/0561c78c9dcb5876a140ff7be97f0f37abc36ec8/xds/internal/balancer/clusterimpl/clusterimpl.go#L208
- In the above function, the LB policy
  - initially checks for some error conditions
  - applies new load reporting configuration, if any
  - switches to a new child policy, if the child policy name has changed
  - pushes the newly received load balancer configuration onto a channel here: https://github.com/grpc/grpc-go/blob/0561c78c9dcb5876a140ff7be97f0f37abc36ec8/xds/internal/balancer/clusterimpl/clusterimpl.go#L253
  - and returns from `UpdateClientConnState`
- The newly received configuration update is processed here asynchronously: https://github.com/grpc/grpc-go/blob/0561c78c9dcb5876a140ff7be97f0f37abc36ec8/xds/internal/balancer/clusterimpl/clusterimpl.go#L475
- And a new picker is pushed upwards here: https://github.com/grpc/grpc-go/blob/0561c78c9dcb5876a140ff7be97f0f37abc36ec8/xds/internal/balancer/clusterimpl/clusterimpl.go#L479

This needs to be changed to return a picker synchronously upon receipt of a configuration update.

### PR DESCRIPTION
**Pull request by aranjans**
### Problem
[#5469](https://github.com/grpc/grpc-go/issues/5469) recommends an audit of existing LB policies to ensure that they update their pickers synchronously upon receipt of a configuration update. `cluster_resolver` neither does update its picker synchronously nor wait for the child policy configuration update inline.

### What does this PR do?
This PR ensures that every time configuration update is triggered, it waits for child policy configuration update inline.

RELEASE NOTES: 

- cluster_resolver: wait for child policy configuration update inline

### CODE REVIEW

File: xds/internal/balancer/clusterresolver/clusterresolver.go [hunk 1]
```diff
@@ -73,11 +77,10 @@ func (bb) Build(cc balancer.ClientConn, opts balancer.BuildOptions) balancer.Bal
 	}
 
 	b := &clusterResolverBalancer{
-		bOpts:    opts,
-		updateCh: buffer.NewUnbounded(),
-		closed:   grpcsync.NewEvent(),
-		done:     grpcsync.NewEvent(),
-
+		bOpts:                opts,
```
Comment (purnesh42H) [inline xds/internal/balancer/clusterresolver/clusterresolver.go:80]: did anything change here? Can't figure out the change in line 80-83

File: xds/internal/balancer/clusterresolver/clusterresolver.go [hunk 2]
```diff
@@ -306,6 +310,9 @@ func (b *clusterResolverBalancer) run() {
 			switch update := u.(type) {
 			case *ccUpdate:
 				b.handleClientConnUpdate(update)
+				if update.done != nil {
```
Comment (purnesh42H) [inline xds/internal/balancer/clusterresolver/clusterresolver.go:313]: nit: add inline comment for the purpose of this if condition

Comment (aranjans) [reply xds/internal/balancer/clusterresolver/clusterresolver.go:314]: Updated

File: xds/internal/balancer/clusterresolver/clusterresolver.go (hunk 1 above)
Comment (aranjans) [reply xds/internal/balancer/clusterresolver/clusterresolver.go]: There was a line break earlier, i believe it got removed. Updated the PR now.

File: xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go [hunk 3]
```diff
@@ -445,3 +446,64 @@ func (s) TestOutlierDetectionConfigPropagationToChildPolicy(t *testing.T) {
...
+	// Create a listener to be used by the management server. The test will
+	// close this listener to simulate ADS stream breakage.
+	lis, err := testutils.LocalTCPListener()
+	if err != nil {
+		t.Fatalf("testutils.LocalTCPListener() failed: %v", err)
+	}
+
+	// Start an xDS management server with the above restartable listener, and
+	// push a channel when the stream is closed.
+	//streamClosedCh := make(chan struct{}, 1)
```
Comment (purnesh42H) [inline xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go:467]: why is this commented?

File: xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go [hunk 4]
```diff
@@ -445,3 +446,64 @@ func (s) TestOutlierDetectionConfigPropagationToChildPolicy(t *testing.T) {
...
+
+	client := testgrpc.NewTestServiceClient(cc)
+	if _, err := client.EmptyCall(ctx, &testpb.Empty{}); err != nil {
+		t.Fatalf("EmptyCall() failed: %v", err)
+	}
+
+	// Close the listener and ensure that the config is updated.
+	lis.Close()
+	select {
+	case <-configUpdated:
```
Comment (purnesh42H) [inline xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go:505]: how do we verify if update was synchronous? Do we need to check for `NewConfigHook`

Comment (aranjans) [reply xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go:505]: So if you look at the top of the test, we are overriding `NewConfigHook` where we are pushing a struct into `configUpdated` channel. And at the end we are actually waiting for that channel which will ensure that update was synchronous.

**Pull request closed without merging**